lib/thrift/protocols/binary.ex

```diff
@@ -53,10 +49,9 @@ defmodule Thrift.Protocols.Binary do
   def serialize(_, nil) do
     []
   end
-  def serialize(:bool, value) do
-    value = bool_to_int(value)
-    <<value::8-signed>>
-  end
+  def serialize(:bool, nil),   do: <<0::8-signed>>
```

Comment:
If a boolean field in a struct is nil, that field would be omitted. That logic isn't represented in this generic serialization because it doesn't handle structs. It passes them off to presumably-generated struct serializers. (See `serialize(:struct, _)` further down.)

If you serialize a list of bools, we would treat a nil in that list as falsey.

```elixir
serialize({:list, :bool}, [nil, false, true]) |> :erlang.iolist_to_binary
# <<2, 0, 0, 0, 3, 0, 0, 1>>
#   |  \--------/  |  |  \- true
#   |      |       |  \---- false
#   |      |       \------- nil
#   |      \--------------- length = 3
#   \---------------------- element type = bool
```

Note that this doesn't currently work because of the `serialize(_, nil)` clause further up.